### PR TITLE
feat: add hangman print statements

### PR DESCRIPTION
**Pull request by h0uter**
What is recommended way to specificy this pull request should be merged after the `ft-add-lives` one?

### CODE REVIEW

**Comment by bobluppes** [comment]
> What is recommended way to specificy this pull request should be merged after the `ft-add-lives` one?

@h0uter, I think that merge order makes sense. I would then also change the target branch of this PR from `main` to `ft-add-lives` such that the commits of your base branch do not show up in the diffs here (I think only PR authors can edit this).

edit: turns out I actually could edit this

File: src/lib.rs [hunk 1]
```diff
@@ -1,7 +1,10 @@
 use std::error::Error;
 use std::io;
 
-fn get_user_input(prompt: &str) -> String {
+// CLI input output
+
+fn take_user_input(prompt: &str) -> String {
```
Comment (bobluppes) [inline src/lib.rs:6]: Out of scope for this pr: I think such "section" comments between function definitions is a strong sign that there may be value in extracting functions to their own class / module.

File: src/lib.rs [hunk 2]
```diff
@@ -13,21 +16,93 @@ fn get_user_input(prompt: &str) -> String {
...
+           |
+     =========",
+];
+
+fn print_hangman_stage(incorrect_guesses: usize) {
+    if incorrect_guesses < HANGMAN_STAGES.len() {
+        println!("{}", HANGMAN_STAGES[incorrect_guesses]);
+    } else {
+        panic!("Invalid number of incorrect guesses.");
+    }
```
Comment (bobluppes) [inline src/lib.rs:90]: This introduces an  implicit coupling between the number of lives a player has in the `config` and the number of drawing stages. Currently this is shown by the fact that you check whether lives/incorrect guesses do not exceed a magic number both here and in lines 186 - 188.

What do you think about being more "permissive" and simply always draw the last hangman stage if `incorrect_guesses > HANGMAN_STAGES.len()`? Alternatively, if this coupling is really desired, you could think about making it more explicit by getting rid of `config.lives` and simply determining whether the player is dead by checking if they reached the final hangman stage.

File: src/lib.rs [hunk 3]
```diff
@@ -13,21 +16,93 @@ fn get_user_input(prompt: &str) -> String {
...
+    wrong_guesses: usize,
 }
 
 fn run_game_loop(config: &mut Config, guess: char, player_state: &mut PlayerState) -> GameState {
     if !config.secret_word.contains(guess) {
         player_state.wrong_guesses += 1;
 
-        if player_state.wrong_guesses > config.lives {
+        print_hangman_stage(player_state.wrong_guesses + (6 - config.lives));
+
```
Comment (bobluppes) [inline src/lib.rs:104]: [nit]: We could think about simplifying this if we set the number of lives to 6:
```suggestion
        print_hangman_stage(player_state.wrong_guesses);

```

**Comment by h0uter** [comment]
> > What is recommended way to specificy this pull request should be merged after the `ft-add-lives` one?
> 
> @h0uter, I think that merge order makes sense. I would then also change the target branch of this PR from `main` to `ft-add-lives` such that the commits of your base branch do not show up in the diffs here (I think only PR authors can edit this).
> 
> edit: turns out I actually could edit this

and then how do we merge it after that?

**Review by bobluppes — COMMENTED** [review]
Very cool! I never expected this project to have graphics and animations 🤩 

What do you think about the comments below, I wonder whether we could remove the coupling between the max number of player lives and the number of hangman states.

**Comment by bobluppes** [comment]
> and then how do we merge it after that?

@h0uter you would have to merge the PRs manually in-order. So in this case that would mean first merging #4 into `main`, afterwards github automatically adjusts the target of this branch and we can merge it.

**Pull request merged**